src/frequenz/client/assets/types.py

```diff
@@ -12,125 +12,214 @@
+class Battery:
+    """A wrapper class for the protobuf Battery message."""
 
-    A delivery area is a geographical area that is served by a microgrid.
-    """
+    type: BatteryType
+    """The type of battery."""
 
-    code: str
-    code_type: str
 
-    @staticmethod
-    def from_protobuf(pb: PBDeliveryArea) -> "DeliveryArea":
-        """
-        Create a DeliveryArea instance from a protobuf message.
-
-        Args:
-            pb: The protobuf DeliveryArea message.
-
-        Returns:
-            A new DeliveryArea instance populated with data from the protobuf message.
-        """
-        return DeliveryArea(
-            code=pb.code,
-            code_type=str(pb.code_type),
-        )
+@dataclass(frozen=True)
+class EvCharger:
+    """A wrapper class for the protobuf EvCharger message."""
+
+    type: EvChargerType
+    """The type of EV charger."""
 
 
 @dataclass(frozen=True)
-class Location:
-    """A wrapper class for the protobuf Location message.
+class GridConnectionPoint:
+    """A wrapper class for the protobuf GridConnectionPoint message."""
 
-    A location is a geographical location that is served by a microgrid.
-    """
+    rated_fuse_current: int
+    """The rated fuse current of the grid connection point."""
 
-    latitude: float
-    longitude: float
-    country_code: str | None
 
-    @staticmethod
-    def from_protobuf(pb: PBLocation) -> "Location":
-        """Create a Location instance from a protobuf message.
-
-        Args:
-            pb: The protobuf Location message.
-
-        Returns:
-            A new Location instance populated with data from the protobuf message.
-        """
-        return Location(
-            latitude=pb.latitude,
-            longitude=pb.longitude,
-            country_code=pb.country_code,
-        )
+@dataclass(frozen=True)
+class Inverter:
+    """A wrapper class for the protobuf Inverter message."""
+
+    type: InverterType
+    """The type of inverter."""
 
 
 @dataclass(frozen=True)
-class Microgrid:
-    """A wrapper class for the protobuf Microgrid message.
+class PowerTransformer:
+    """A wrapper class for the protobuf PowerTransformer message."""
+
+    primary: float
+    """The primary of the power transformer."""
+
+    secondary: float
+    """The secondary of the power transformer."""
+
 
-    A microgrid is a localized group of electricity sources and loads that normally
-    operates connected to and synchronous with the traditional wide area electrical
-    grid (macrogrid), but is able to disconnect from the grid and operate autonomously.
+ElectricalComponentCategorySpecificInfo = (
+    Battery | EvCharger | GridConnectionPoint | Inverter | PowerTransformer
+)
+
+
+@dataclass(frozen=True)
+class Lifetime:
+    """A wrapper class for the protobuf Lifetime message."""
+
+    start_time: datetime
+    """The start time of the operational lifetime."""
+
+    end_time: datetime
+    """The end time of the operational lifetime."""
```

Comment:
These should definitely support `None`, as timestamps are composed messages in protobuf so the field could be missing. It is also specified in the protobuf message what is the meaning of a missing field.